dev/tests/integration/testsuite/Magento/LoginAsCustomer/_files/admin_rollback.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+use Magento\Authorization\Model\RoleFactory;
+use Magento\Authorization\Model\Role;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\User\Model\User;
+use Magento\Authorization\Model\RulesFactory;
+use Magento\Authorization\Model\Rules;
+
+//Deleting the user and the role.
+/** @var User $user */
+$user = Bootstrap::getObjectManager()->create(User::class);
+$user->load('TestAdmin1', 'username');
+$user->delete();
+/** @var Role $role */
+$role = Bootstrap::getObjectManager()->get(RoleFactory::class)->create();
+$role->load('test_custom_role', 'role_name');
+/** @var Rules $rules */
+$rules = Bootstrap::getObjectManager()->get(RulesFactory::class)->create();
+$rules->load($role->getId(), 'role_id');
+$rules->delete();
+$role->delete();
```

Comment:
Please, do not use deprecated methods in your code. (save, delete of ORM models). Use repositories or resource models instead.

Reply:
@rogyar need suggestion for this also. For this also I did follow the same as I can't find any repository.

Reply:
You may create a new instance of the following class using the `ObjectManager`

https://github.com/magento/magento2/blob/67fae82c6278827f2c9b0f63083486a9d7740366/app/code/Magento/Authorization/Model/ResourceModel/Role.php#L13

Then use ::delete method of the newly created resource model (and passing your `role` model as an argument)

Reply:
@rogyar updated the PR with suggested changes. Please review, Thank you